django/utils/inspect.py

```diff
@@ -56,14 +56,22 @@ def get_func_full_args(func):
 
 def func_accepts_kwargs(func):
     """Return True if function 'func' accepts keyword arguments **kwargs."""
-    return any(p for p in _get_callable_parameters(func) if p.kind == p.VAR_KEYWORD)
+    return any(
+        p
+        for p in _get_callable_parameters(func)
+        if p.kind == p.VAR_KEYWORD or p.kind == p.KEYWORD_ONLY
+    )
 
 
 def func_accepts_var_args(func):
     """
     Return True if function 'func' accepts positional arguments *args.
     """
-    return any(p for p in _get_callable_parameters(func) if p.kind == p.VAR_POSITIONAL)
+    return any(
+        p
+        for p in _get_callable_parameters(func)
+        if p.kind == p.VAR_POSITIONAL or p.kind == p.POSITIONAL_ONLY
+    )
```

Comment:
I'm sorry but it appears this patch breaks these 2 functions.

`func_accepts_var_args()` should only return `True` for `VAR_POSITIONAL` - the name of the function and docstrings are clear on this 😊

as for `func_accepts_kwargs()` - the name is a misnomer but the docstrings state the intended purpose. If you look at its usage throughout the codebase though the intention that it only return `True` for `**kwargs`.

Reply:
Thanks. I had to read better before